include/fiction/networks/virtual_pi_network.hpp

```diff
@@ -0,0 +1,505 @@
+                    {
+                        old2new[g] = ntk_dest.create_xor(children[0], children[1]);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_is_maj_v<Ntk> && mockturtle::has_create_maj_v<Ntk>)
+                {
+                    if (ntk.is_maj(g))
+                    {
+                        old2new[g] = ntk_dest.create_maj(children[0], children[1], children[2]);
+                        return true;
+                    }
+                }
+                if constexpr (fiction::has_is_buf_v<Ntk> && mockturtle::has_create_buf_v<Ntk>)
+                {
+                    if (ntk.is_buf(g))
+                    {
+                        old2new[g] = ntk_dest.create_buf(children[0]);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_node_function_v<Ntk> && mockturtle::has_create_node_v<Ntk>)
+                {
+                    old2new[g] = ntk_dest.create_node(children, ntk.node_function(g));
+                    return true;
+                }
+
+                return true;
+            });
+
+        ntk.foreach_po(
+            [this, &ntk_dest, &old2new](const auto& po)
+            {
+                const auto tgt_signal = old2new[ntk.get_node(po)];
+                const auto tgt_po     = ntk.is_complemented(po) ? ntk_dest.create_not(tgt_signal) : tgt_signal;
+
+                ntk_dest.create_po(tgt_po);
+            });
+
+        // restore signal names if applicable
+        fiction::restore_names(ntk, ntk_dest, old2new);
+
+        return ntk_dest;
+    }
+
+  private:
+    using TopoNtkSrc = mockturtle::topo_view<Ntk>;
+    Ntk        ntk;
+    TopoNtkSrc ntk_topo;
+
+    [[nodiscard]] std::pair<Ntk, mockturtle::node_map<mockturtle::signal<Ntk>, Ntk>>
+    initialize_copy_virtual_pi_network(Ntk& src)
+    {
+        static_assert(mockturtle::is_network_type_v<Ntk>, "Ntk is not a network type");
+        static_assert(mockturtle::has_get_constant_v<Ntk>, "Ntk does not implement the get_constant method");
+        static_assert(mockturtle::has_create_pi_v<Ntk>, "Ntk does not implement the create_pi method");
+        static_assert(mockturtle::has_get_node_v<Ntk>, "Ntk does not implement the get_node method");
+        static_assert(fiction::has_foreach_real_pi_v<Ntk>, "Ntk does not implement the foreach_pi method");
+
+        mockturtle::node_map<mockturtle::signal<Ntk>, Ntk> old2new(src);
+        Ntk                                                dest;
+        old2new[src.get_constant(false)] = dest.get_constant(false);
+        if (src.get_node(src.get_constant(true)) != src.get_node(src.get_constant(false)))
+        {
+            old2new[src.get_constant(true)] = dest.get_constant(true);
+        }
+        src.foreach_real_pi([&](auto const& n) { old2new[n] = dest.create_pi(); });
+        return {dest, old2new};
```

Comment:
I find it easier to read when code has more blank lines. This may be a personal preference.